### ISSUE TITLE
iOS Voiceover: article pane feed image lacks indication it opens feed inspector dialog

### ISSUE THREAD

**Issue by ollicle** [Ranchero-Software/NetNewsWire#4591]
## Problem

On iOS, tapping the feed icon when viewing an article opens the feed inspector dialog.

As an item in the UI, the image is announced by Voiceover as “image” with no indication that it can activated.

## Fix

I see an opportunity to improve discovery of this feature, at least for people using VoiceOver.

- Use a `<button>` element that will be appropropriately announced as an operable element in Voiceover
- Provide a meaningful accessible label for the button

## Challenges

- The current implementation enhances an element provided by a theme template. To provide backward compatibility with existing custom themes the solution must also avoid relying on changes to theme HTML.
- Clicking the image on Mac does nothing. Using a button to display the icon only makes sense on iOS.
- A decision on what to label the button as it currently is unnamed? “Feed options for <feed_link_title>” perhaps? 

## Suggested implementation

Fortunately the existing implementation which attaches a click event handler to the `<img id="nnwImageIcon"` using JavaScript in [main_ios.js](https://github.com/Ranchero-Software/NetNewsWire/blob/bc7f9050165cb9bddffc67948cafabb5319030de/iOS/Resources/main_ios.js#L148) demonstrates an approach that will work.

In `showFeedInspectorSetup()`:

1. Find the element with id `nnwImageIcon`
2. Wrap it in a button element (with a class to enable theming)
3. Apply the click handler to the button to showFeedInspector instead of the img
4. Find the element with class `feedlink`, if it exists append its textContent to “Feed options for ” to construct a label (else use “Feed options”)
5. Assign the constructed label to the img alt attribute
6. In core.css add theme styles to suppress default button styles (e.g. `display:contents;`).

## Aside

The “image” annoucement of the favicon image provides no value to a user of NetNewsWire on MacOS. Using the feed name again here would be redundant.

Same goes for iOS when JavaScript is not enabled, annoucing “image” provides no value.

Adding an empty alt as a default `<img id="nnwImageIcon" alt=""` will suppress the announcement.

I propose doing so in shipped NetNewsWire templates will be an enhancement in both scenarios.

**Comment by ollicle** [comment]
A second thought on this one. On the surface this appears to be purely an accessibility issue, however it strays into usability when you consider that the feed options dialog is a somewhat hidden feature for everyone – regardless of their chosen mode of use.

Could a theme be an appropriate venue to revisit this decision?

I’m thinking, before wrapping the nnwImageIcon img with a button, the JavaScript could check the template for the existence of a dedicated button using an appropriate id. So a theme could incorporate something like:

```
<button id="nnwOpenFeedOptions" hidden>Feed options for [[feed_link_title]]</button>
```

NetNewsWire then js removes the hidden attribute and applies the click listener to this button. Otherwise enhancing the img to a button istead.

Interested in your thoughts.

**Comment by brentsimmons** [comment]
Fixed in 7.1.3.
